core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -424,6 +438,10 @@ private void installConfig(PluginInfo info, Path tmpConfigDir, Path destConfigDi
         // create the plugin's config dir "if necessary"
         Files.createDirectories(destConfigDir);
 
+        final PosixFileAttributes destConfigDirAttributes =
+                Files.getFileAttributeView(destConfigDir.getParent(), PosixFileAttributeView.class).readAttributes();
+        setOwnerGroup(destConfigDir, destConfigDirAttributes);
```

Comment:
> If the umask is set wrong, the onus is on the user.

If we're installed by rpm or deb then all of our tools need to follow whatever rules we've set for ourselves regarding permissions and ownership. If we can work around this at the rpm or deb level I'm fine with that but ultimately we can't just not do it because permissions are icky. And they are icky and slow to test and before this PR we weren't testing these particular permissions properly.

IIRC the reason for this choice of permissions is that these configs can contain sensitive stuff so they shouldn't be world readable. They need to be owned by root so the Elasticsearch process can't edit them.

All that being said, if we made the parent config folder's ownership `root:elasticsearch` and we gave it `0750` would that allow us to not have this code again? I don't see a problem with that ownership is acceptable, even if it is a bit weird. I see only `cups` has non `root:root` ownership in `/etc`. So I'm 95% sure it'd violate someone's package rules, maybe for a good reason I haven't thought of or maybe for a silly reason. I admit I'm not that good at this stuff so, yeah.

I guess I proposed a potential work around. I still think we should merge as is because it gets us much much closer to not being broken. If we want to remove this code and use different ownership on `/etc/elasticsearch` then I think we can do that in a followup so it's potential controversy doesn't get mixed into this.